server/src/main/java/io/crate/execution/dsl/projection/ProjectionType.java

```diff
@@ -44,11 +44,14 @@ public enum ProjectionType {
     EVAL(EvalProjection::new),
     PROJECT_SET(ProjectSetProjection::new),
     WINDOW_AGGREGATION(WindowAggProjection::new),
-    TOPN_DISTINCT(TopNDistinctProjection::new);
+    TOPN_DISTINCT(TopNDistinctProjection::new),
+    CORRELATED_JOIN(in -> {
+        throw new UnsupportedOperationException("Cannot stream correlated join projection");
```

Comment:
Because the projection contains planner components and the logical plan. Those are all not streamable.

Correlated join execution is limited to the handler node.